scrapy/extensions/feedexport.py

```diff
@@ -146,6 +146,32 @@ def _store_in_thread(self, file):
             key.close()
 
 
+
+class GCSFeedStorage(BlockingFeedStorage):
+
+    project_id = None
```

Comment:
It doesn't seem right to store this information as class attributes. 
You should be defining those values in the init method, don't you?

Reply:
You mean to make it private?
I don't see any difference in:
```
class A:
   attribute = None
   def __init__(self, a):
        self.attribute = a

class A:
   def __init__(self, a):
        self.attribute = a
```

Actually, I prefer the first one because it makes things explicit (there is an attribute called `attribute`).

Reply:
Never mind. Mixed some concepts here.

Reply:
No, I mean to declare them as instance attributes instead of class attributes.
This is how we are doing in `S3FeedStorage`, for example.

Class attributes are really useful when storing constants or helper values, things that are shared across multiple instances of the same class.

In this case, I don't think you should reuse the storage settings.

In worst case scenarios, it could also lead to indesirable mutability when trying to change an instance attribute value causes the class attribute value and therefore all other instance attribute values to be changed too.

In a nutshell, you **need** those values in order to initialize the instance, so there's no reason to keep an unused default value as a class attribute.